From 268d510ad47098b4800ac269f192486cd6f43dc8 Mon Sep 17 00:00:00 2001
From: dengliqiang <liqiangx.deng@intel.com>
Date: Thu, 26 Sep 2019 19:58:42 +0800
Subject: [PATCH 7/7] make cpu and gpu instance run in parallel

need two rc files, one for cpu and another for gpu

Tracked-On: OAM-86770
Signed-off-by: dengliqiang <liqiangx.deng@intel.com>
---
 intel_nn_hal/Android.mk                                       | 3 ++-
 ....rc => android.hardware.neuralnetworks@1.1-generic-cpu.rc} | 0
 .../android.hardware.neuralnetworks@1.1-generic-gpu.rc        | 4 ++++
 3 files changed, 6 insertions(+), 1 deletion(-)
 rename intel_nn_hal/{android.hardware.neuralnetworks@1.1-generic.rc => android.hardware.neuralnetworks@1.1-generic-cpu.rc} (100%)
 create mode 100644 intel_nn_hal/android.hardware.neuralnetworks@1.1-generic-gpu.rc

diff --git a/intel_nn_hal/Android.mk b/intel_nn_hal/Android.mk
index 68d726e..547623c 100644
--- a/intel_nn_hal/Android.mk
+++ b/intel_nn_hal/Android.mk
@@ -81,7 +81,8 @@ include $(BUILD_SHARED_LIBRARY)
 ###############################################################
 include $(CLEAR_VARS)
 LOCAL_MODULE := android.hardware.neuralnetworks@1.1-generic-service
-LOCAL_INIT_RC := android.hardware.neuralnetworks@1.1-generic.rc
+LOCAL_INIT_RC := android.hardware.neuralnetworks@1.1-generic-cpu.rc \
+    android.hardware.neuralnetworks@1.1-generic-gpu.rc
 LOCAL_MODULE_RELATIVE_PATH := hw
 LOCAL_PROPRIETARY_MODULE := true
 LOCAL_MODULE_OWNER := intel
diff --git a/intel_nn_hal/android.hardware.neuralnetworks@1.1-generic.rc b/intel_nn_hal/android.hardware.neuralnetworks@1.1-generic-cpu.rc
similarity index 100%
rename from intel_nn_hal/android.hardware.neuralnetworks@1.1-generic.rc
rename to intel_nn_hal/android.hardware.neuralnetworks@1.1-generic-cpu.rc
diff --git a/intel_nn_hal/android.hardware.neuralnetworks@1.1-generic-gpu.rc b/intel_nn_hal/android.hardware.neuralnetworks@1.1-generic-gpu.rc
new file mode 100644
index 0000000..84ad7a7
--- /dev/null
+++ b/intel_nn_hal/android.hardware.neuralnetworks@1.1-generic-gpu.rc
@@ -0,0 +1,4 @@
+service neuralnetworks-hal-1-1-gpu /vendor/bin/hw/android.hardware.neuralnetworks@1.1-generic-service -D GPU
+    class hal
+    user system
+    group system
-- 
2.23.0

